lib/Doctrine/ORM/EntityRepository.php

```diff
@@ -225,7 +225,8 @@ public function __call($method, $arguments)
 
             default:
                 throw new \BadMethodCallException(
-                    "Undefined method '$method'. The method name must start with ".
+                    "In Repository for Entity " . $this->_entityName . 
+                    " Undefined method '$method'. The method name must start with ".
                     "either findBy or findOneBy!"
```

Comment:
I would like something more in this direction, as it's more verbose:
```php
throw new \BadMethodCallException(
    "Undefined method called: '".get_class($this)."::{$method}()'. The ".
    "repository method name must start with either 'findBy' or 'findOneBy'."
);
```
This would produce "Undefined method called: 'Foo\Bar\BazRepository::getAll()'. The repository method name must start with either 'findBy' or 'findOneBy'."

*Note that I left out the exclamation mark on purpose*